docs/data_driven_testing.adoc

```diff
@@ -289,3 +288,35 @@ include::{sourcedir}/datadriven/DataSpec.groovy[tag=unrolled-3a]
     ...
 include::{sourcedir}/datadriven/DataSpec.groovy[tag=unrolled-3b]
 ----
+
+Additionally to the data variables the tokens `#featureName` and `#iterationIndex` are supported.
+The former does not make much sense inside an actual feature name, but there are two other places
+where an unroll pattern can be defined, where it is more useful.
+
+[source,groovy,indent=0]
+----
+def "#person is #person.age years old [iterationIndex: #iterationIndex]"() {
+----
+
+Alternatively to specifying the unroll pattern as method name, it can be given as parameter
+to the `@Unroll` annotation which takes precedence over the method name:
+
+[source,groovy,indent=0]
+----
+@Unroll("#featureName[#iterationIndex] (#person.name is #person.age years old)")
+def "person age should be calculated properly"() {
+----
+
+If neither a parameter to the annotation is given, nor the method name contains a `#`,
+the system property `spock.globalUnrollPattern` is inspected. If it is set to a non-blank
+string, this value is used as unroll pattern. This could for example be set to
+
+- `#featureName` to have all iterations reported with the same name, or
```

Comment:
Won't it hit the problem with [#1070](https://github.com/spockframework/spock/issues/1070#issuecomment-573421786)?

Reply:
No, that seems to be a problem in combination with vintage engine.
In the current state data-driven methods are reported as the individual iterations but with the literal feature name without name unrolling unless you add the `@Unroll` annotation or use your extension (next PR will fix this). Execute any data-driven test Spock itself and you will see that the iterations are reported fine in IJ and in the Gradle HTML report with the same name.